### PR TITLE
fix: Focus switches on new Messages (ACC-310)

### PR DESCRIPTION
**Pull request by przemvs**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-310" title="ACC-310" target="_blank">BugACC-310</a>  [Web] Focus switches on new Messages
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: src/script/hooks/useConversationFocus.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
```
Comment (thisisamir98) [inline src/script/hooks/useConversationFocus.ts]:
```suggestion
 * Copyright (C) 2023 Wire Swiss GmbH
```

**Review by thisisamir98 — APPROVED** [review]

File: src/script/components/list/ConversationListCell.tsx [hunk 2]
```diff
@@ -167,15 +144,15 @@ const ConversationListCell: React.FC<ConversationListCellProps> = ({
     }
   };
 
+  const isFocused = currentFocusId === conversation.id;
```
Comment (PatrykBuniX) [inline src/script/components/list/ConversationListCell.tsx]: What about passing `isFocused` to this component? I don't think it needs to know what `currentFocusId` is, but rather if its focused or not.

File: src/script/page/LeftSidebar/panels/Conversations/Conversations.tsx [hunk 3]
```diff
@@ -206,9 +202,6 @@ const Conversations: React.FC<ConversationsProps> = ({
           data-uie-name="status-name"
           role="presentation"
           tabIndex={TabIndex.FOCUSABLE}
-          // personal user won't see availability status menu, on blur of the userName
```
Comment (arjita-mitra) [inline src/script/page/LeftSidebar/panels/Conversations/Conversations.tsx:209]: why did you remove this code?

// personal user won't see availability status menu, on blur of the userName
          // conversation list should get the focus
          onBlur={() => focusConversationList(true)}

File: src/script/components/list/ConversationListCell.tsx [hunk 4]
```diff
@@ -167,15 +144,15 @@ const ConversationListCell: React.FC<ConversationListCellProps> = ({
...
+
   // always focus on the selected conversation when the folder tab loaded
```
Comment (arjita-mitra) [inline src/script/components/list/ConversationListCell.tsx:147]: /
Screenshot 2023-02-03 at 14 45 39
/ always focus on the selected conversation when the folder tab loaded.

This condition is not working correctly anymore. Attached a screenshot where new group is focused when folder tab opened but the selected conversation is group1.

File: src/script/page/LeftSidebar/panels/Conversations/Conversations.tsx (hunk 3 above)
Comment (przemvs) [reply src/script/page/LeftSidebar/panels/Conversations/Conversations.tsx:209]: Now first item of conversation always have a tabIndex = 0. So onBlur method is unnecessary. :)

**Review by PatrykBuniX — APPROVED** [review]

**Review by arjita-mitra — CHANGES_REQUESTED** [review]

File: src/script/components/list/ConversationListCell.tsx (hunk 4 above)
Comment (przemvs) [reply src/script/components/list/ConversationListCell.tsx:147]: Consulted with Rahil, We always focus from first conversation if we using Tab or Shift + Tab, also in Folders View and Conversations.

**Review by arjita-mitra — APPROVED** [review]

**Pull request merged**